src/Symfony/Component/Config/ConfigCache.php

```diff
@@ -69,28 +70,61 @@ public function getPath()
      * cache file exists.
      *
      * @return bool true if the cache is fresh, false otherwise
+     *
+     * @deprecated since version 2.8, to be removed in 3.0.
      */
     public function isFresh()
     {
-        if (!is_file($this->file)) {
-            return false;
-        }
+        @trigger_error(__NAMESPACE__.'\ConfigCache::isFresh() is deprecated since version 2.8 and will be removed in 3.0. Use the isValid() method instead and pass the appropriate MetadataValidators, or even better use a \Symfony\Component\Config\ConfigCacheFactoryInterface implementation to create and validate the cache.', E_USER_DEPRECATED);
 
-        if (!$this->debug) {
+        if (!$this->debug && is_file($this->file)) {
             return true;
         }
 
+        return $this->isValid(array(new ResourceValidator()));
+    }
+
+    /**
+     * Use MetadataValidators to check if the cache is still valid.
+     *
+     * The first MetadataValidator that supports a given resource is considered authoritative.
+     * Resources with no matching MetadataValidators will silently be ignored and considered fresh.
+     *
+     * This method <em>does not</em> take the debug flag into consideration: Whether or not a cache
+     * should be checked in production mode and/or which validators need to be applied is a decision
+     * left to the client of this method.
+     *
+     * @param MetadataValidatorInterface[] $validators List of validators the metadata is checked against.
+     *                                                 The first validator that supports a resource is considered authoritative.
+     *
+     * @return bool True if all supported resources and valid, false otherwise
+     */
+    public function isValid(array $validators = null)
+    {
+        if (!is_file($this->file)) {
+            return false;
+        }
+
         $metadata = $this->getMetaFile();
         if (!is_file($metadata)) {
-            return false;
+            return true;
         }
 
         $time = filemtime($this->file);
         $meta = unserialize(file_get_contents($metadata));
+
         foreach ($meta as $resource) {
-            if (!$resource->isFresh($time)) {
-                return false;
+            foreach ($validators as $validator) {
+                if (!$validator->supports($resource)) {
+                    continue; // next validator
+                }
+                if ($validator->isFresh($resource, $time)) {
+                    continue; // no need to further check this resource
```

Comment:
this should be a break, not a continue. Otherwise you are checking the following validators on this resource